File: backend/src/main/java/proyecto/service/ProductoService.java
package proyecto.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import proyecto.entities.ProductoEntity;
import proyecto.repositories.CategoriaRepository;
import proyecto.repositories.ProductoRepository;

import java.util.List;

@Service
public class ProductoService {

    @Autowired
    ProductoRepository productoRepository;

    @Autowired
    CategoriaRepository categoriaRepository;

    public List<ProductoEntity> findAll(){
        return productoRepository.findAll();
    }

    public ProductoEntity findById(int id_producto){
        return productoRepository.findById(id_producto);
    }

    public ProductoEntity create(ProductoEntity producto){
        return productoRepository.create(producto);
    }

    public ProductoEntity update(ProductoEntity producto){
        return productoRepository.update(producto);
    }

    public boolean delete(int id_producto){
        return productoRepository.delete(id_producto);
    }

    public List<ProductoEntity> findByCategoria(int id_categoria){
        return productoRepository.findByCategoria(id_categoria);
    }





}

File: front-end/src/services/productoService.js
import {httpClient} from './httpCommon';

const getAll = () => {
    return httpClient.get("/api/producto");
}

const findByCategoria = idCategoria => {
    return httpClient.get(`/api/producto/categoria/${idCategoria}`);
}

const findById = id => {
    return httpClient.get(`/api/producto/${id}`);
}

const create = data => {
    return httpClient.post("/api/producto/create", data);
}

const topProductos = () => {
    return httpClient.get("/api/producto/top");
}

const update = data => {
    return httpClient.put("/api/producto/update", data);
}

const remove = id => {
    return httpClient.delete(`/api/producto/delete/${id}`);
}

export default {
    getAll,
    findByCategoria,
    findById,
    create,
    update,
    remove,
    topProductos
};